packages/@react-native-windows/cli/src/test/dependencyConfig.test.ts

```diff
@@ -0,0 +1,141 @@
+    // Tests the result given a windows project config in react-native.config.js
+    const userConfig: Partial<WindowsDependencyConfig> =
+      rnc.dependency.platforms.windows;
+    let expectedConfig: WindowsDependencyConfig | null = rnc.expectedConfig;
+
+    if (expectedConfig !== null) {
+      expectedConfig.folder = folder;
+    }
+
+    expect(dependencyConfigWindows(folder, userConfig)).toEqual(expectedConfig);
+  });
+
+  test(`dependencyConfig - ${testLibName} (Ignore react-native.config.js)`, () => {
+    // Tests the result of ignoring the windows project config in react-native.config.js
+    const userConfig: Partial<WindowsDependencyConfig> = {};
+    const expectedConfig: WindowsDependencyConfig | null =
+      rnc.expectedConfigIgnoringOverride;
+
+    if (expectedConfig !== null) {
+      expectedConfig.folder = folder;
+    }
+
+    expect(dependencyConfigWindows(folder, userConfig)).toEqual(expectedConfig);
+  });
+}
+
+// Nothing but a react-native.config.js, with nulls
+projectTests('BlankLib');
+
+// Nothing but a windows folder
+projectTests('MissingProjectFilesLib', (folder: string) => {
+  const windowsDir = path.join(folder, 'windows');
+  if (!fs.existsSync(windowsDir)) {
+    fs.mkdirSync(windowsDir);
+  }
+});
+
+// New C++ project based on the template
+projectTests('SimpleCppLib', (folder: string) => {
+  const windowsDir = path.join(folder, 'windows');
+  if (!fs.existsSync(windowsDir)) {
+    fs.mkdirSync(windowsDir);
+  }
+
+  const replacements = {
+    name: 'SimpleCppLib',
+    namespace: 'SimpleCppLib',
+    useMustache: true,
+    projectGuidUpper: testProjectGuid,
+    projectGuidLower: testProjectGuid.toLowerCase(),
+  };
+
+  void copyAndReplace(
+    path.join(templateRoot, 'cpp-lib/proj/MyLib.sln'),
+    path.join(windowsDir, 'SimpleCppLib.sln'),
+    replacements,
+    null,
+  );
+
+  const projDir = path.join(windowsDir, 'SimpleCppLib');
+  if (!fs.existsSync(projDir)) {
+    fs.mkdirSync(projDir);
+  }
+  void copyAndReplace(
+    path.join(templateRoot, 'cpp-lib/proj/MyLib.vcxproj'),
+    path.join(projDir, 'SimpleCppLib.vcxproj'),
+    replacements,
+    null,
+  );
+});
+
+// New C# project based on the template
+projectTests('SimpleCSharpLib', (folder: string) => {
+  const windowsDir = path.join(folder, 'windows');
+  if (!fs.existsSync(windowsDir)) {
+    fs.mkdirSync(windowsDir);
+  }
+
+  const replacements = {
+    name: 'SimpleCSharpLib',
+    namespace: 'SimpleCSharpLib',
+    useMustache: true,
+    projectGuidUpper: testProjectGuid,
+    projectGuidLower: testProjectGuid.toLowerCase(),
+  };
+
+  void copyAndReplace(
+    path.join(templateRoot, 'cs-lib/proj/MyLib.sln'),
+    path.join(windowsDir, 'SimpleCSharpLib.sln'),
+    replacements,
+    null,
+  );
+
+  const projDir = path.join(windowsDir, 'SimpleCSharpLib');
+  if (!fs.existsSync(projDir)) {
+    fs.mkdirSync(projDir);
+  }
+  void copyAndReplace(
```

Comment:
Do these need to be awaited instead of using the void operator? You can have async tests with Jest, and that all works out fine.

Reply:
Oh, looking at things, this works because the only async bit currently is the prompt. I think await would be needed if other async code was added to copyAndReplace.

Reply:
Fixed async issues.